Bottom: 4681cfa1e37bb0c0291148a59bc9e3660b706361
Top:    dc38ee19ceaac265b8c4bf884db9e9a694a401c2
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-12-04 20:31:11 +0000

Make sanity checks in MoveLayerToGroup asserts.

The only caller which uses the returned group value was in the Lesstif HID,
but it did not check for the possibility of a -1 return. The -1 return was
made in the case of a precondition failure, having been passed an out-of
bound layer. (Which represents a software bug).

Remove the return -1 in favour of an assert, and introduce a similar assert
checking the validity of the group number - rather than turning the function
into a NOP if an invalid group was passed.

This is a step towards making the layer and group numbers unsigned, as
we cannot return -1 when that is done.


---

diff --git a/src/misc.c b/src/misc.c
index 3fd9dc6..643730e 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -2183,14 +2183,15 @@ MoveLayerToGroup (int layer, int group)
 {
   int prev, i, j;
 
-  if (layer < 0 || layer >= max_copper_layer + 2)
-    return -1;
+  assert (layer >= 0 && layer < max_copper_layer + 2);
+  assert (group >= 0 && group < max_group);
+
   prev = GetLayerGroupNumberByNumber (layer);
   if ((layer == bottom_silk_layer
         && group == GetLayerGroupNumberByNumber (top_silk_layer))
       || (layer == top_silk_layer
           && group == GetLayerGroupNumberByNumber (bottom_silk_layer))
-      || (group < 0 || group >= max_group) || (prev == group))
+      || prev == group)
     return prev;
 
   /* Remove layer from prev group */
